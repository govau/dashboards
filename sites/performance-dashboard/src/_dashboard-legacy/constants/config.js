export const SCREEN_SM = 768;
export const DEFAULT_HERO_HEIGHT = 240;
