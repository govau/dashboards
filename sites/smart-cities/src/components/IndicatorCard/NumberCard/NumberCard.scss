@import '~style/main';

.wrapper {
  @include shadow;

  display: flex;
  align-items: stretch;
  position: relative; // .indicatorTypeMark is absolute, relative to this
  background: $color-white;
  border-bottom-style: solid;
  border-radius: 4px;

  &.largeCard {
    flex-direction: column;
  }
}

.smallCard {
  height: 80px;
  padding: 4px 8px 7px;
  border-bottom-width: 6px;
}

.largeCard {
  height: 160px;
  padding: 4px 9px 13px;
  border-bottom-width: 8px;
}

.indicatorName {
  flex: 1;
  line-height: 1.4; // fit four lines on a card
  font-weight: 500;

  .smallCard & {
    margin-right: 4px;
    font-size: 12px;
  }

  .largeCard & {
    padding-right: 16px;
    font-size: 16px;
    line-height: 1.1;
  }
}

.indicatorTypeMark {
  position: absolute;

  .smallCard & {
    top: 1px;
    right: 4px;
  }

  .largeCard & {
    top: 2px;
    right: 4px;
  }
}

// do I still need body and wrapper
.indicatorBody {
  margin-left: auto; // push right
  align-self: flex-end; // push down
  line-height: 1;
}

.numberWrapper {
  display: flex;
  align-items: flex-end;
  justify-content: flex-end;
}

.number {
  font-weight: 300;

  .smallCard & {
    font-size: 44px;
  }

  .largeCard & {
    font-size: 90px;
    font-weight: 200;
  }

  .noData & {
    font-size: 24px;
    letter-spacing: -0.6px;
  }
}

.symbol {
  font-weight: 300;

  .smallCard & {
    font-size: 24px;
    margin-bottom: 2px;
  }

  .largeCard & {
    font-size: 44px;
    margin-bottom: 4px;
  }
}

.prefix {
  margin-right: 1px;
}

.suffix {
  margin-left: 1px;
}
