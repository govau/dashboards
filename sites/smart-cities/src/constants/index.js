export * from './misc';
export { default as INDICATORS } from './indicators';
export * from './categories';
export { default as CITIES } from './cities';
