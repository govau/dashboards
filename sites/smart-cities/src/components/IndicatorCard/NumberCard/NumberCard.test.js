import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import NumberCard from './NumberCard';
import { CARD_SIZES } from '../../../constants';

jest.mock('../../IndicatorIcon/IndicatorIcon', () => 'IndicatorIcon');

const defaultProps = {
  city: {
    id: 'perth',
    indicators: {
      population: 10,
    },
    name: 'Perth',
  },
  colorName: 'jobs',
  indicator: {
    id: 'population',
    name: 'Population',
    cardPrefix: 'The',
    cardSuffix: 'per annum',
    format: '0[.]0%',
    contextual: false,
  },
  color: '#fff',
  size: CARD_SIZES.LARGE,
};

it('should match Snapshot', () => {
  const component = renderer.create(<NumberCard {...defaultProps} />);
  expect(component.toJSON()).toMatchSnapshot();
});

it('should apply the largeCard class', () => {
  const component = shallow(
    <NumberCard
      {...defaultProps}
      size={CARD_SIZES.LARGE}
    />
  );

  expect(component.hasClass('largeCard')).toBe(true);
});

it('should apply the smallCard class', () => {
  const component = shallow(
    <NumberCard
      {...defaultProps}
      size={CARD_SIZES.SMALL}
    />
  );

  expect(component.hasClass('smallCard')).toBe(true);
});

it('should show "No data" if there is no data', () => {
  const component = shallow(
    <NumberCard
      {...defaultProps}
      indicator={{
        id: 'activeJourneys', // the city in the test does not have data for this
        name: 'The number of something',
        cardPrefix: '',
        cardSuffix: '',
        format: '',
        contextual: false,
      }}
    />
  );

  expect(component.find('.number').text()).toBe('No data');
});

it('should render no prefix or suffix when none are available', () => {
  const component = shallow(
    <NumberCard
      {...defaultProps}
      indicator={{
        id: 'population',
        name: 'The number of cats',
        cardPrefix: '',
        cardSuffix: '',
        format: '0[.]0',
        contextual: false,
      }}
    />
  );

  expect(component.find('.prefix').length).toBe(0);
  expect(component.find('.suffix').length).toBe(0);
});

it('should render the format prefix', () => {
  const component = shallow(
    <NumberCard
      {...defaultProps}
      indicator={{
        id: 'population',
        name: 'The cost of things',
        cardPrefix: '',
        cardSuffix: '',
        format: '$0[.]0',
        contextual: false,
      }}
    />
  );

  expect(component.find('.prefix').text()).toBe('$');
});

it('should render the card prefix', () => {
  const component = shallow(
    <NumberCard
      {...defaultProps}
      indicator={{
        id: 'population',
        name: 'The cost of things',
        cardPrefix: 'AUD ',
        cardSuffix: '',
        format: '$0[.]0',
        contextual: false,
      }}
    />
  );

  expect(component.find('.prefix').text()).toBe('AUD $');
});

it('should render the format suffix', () => {
  const component = shallow(
    <NumberCard
      {...defaultProps}
      indicator={{
        id: 'population',
        name: 'Population growth',
        cardPrefix: '',
        cardSuffix: '',
        format: '0[.]0%',
        contextual: false,
      }}
    />
  );

  expect(component.find('.suffix').text()).toBe('%');
});

it('should render the card suffix', () => {
  const component = shallow(
    <NumberCard
      {...defaultProps}
      indicator={{
        id: 'population',
        name: 'Population growth',
        cardPrefix: '',
        cardSuffix: '/year',
        format: '0[.]0%',
        contextual: false,
      }}
    />
  );

  expect(component.find('.suffix').text()).toBe('%/year');
});
