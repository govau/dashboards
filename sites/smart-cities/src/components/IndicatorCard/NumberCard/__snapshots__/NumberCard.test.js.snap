// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
<div
  className="wrapper largeCard"
  style={
    Object {
      "borderBottomColor": "#fff",
    }
  }
>
  <p
    className="indicatorName"
  >
    Population
  </p>
  <IndicatorIcon
    className="indicatorTypeMark"
    color="JOBS_500"
    contextual={false}
  />
  <div
    className="indicatorBody"
  >
    <div
      className="numberWrapper"
    >
      <span
        className="prefix symbol"
      >
        The
      </span>
      <span
        className="number"
      >
        1000
      </span>
      <span
        className="suffix symbol"
      >
        %per annum
      </span>
    </div>
  </div>
</div>
`;
